embassy-stm32/src/crc/v1.rs

```diff
@@ -32,20 +32,32 @@ impl<'d> Crc<'d> {
     /// Feeds a word to the peripheral and returns the current CRC value
     pub fn feed_word(&mut self, word: u32) -> u32 {
         // write a single byte to the device, and return the result
+        #[cfg(not(crc_v1))]
+        PAC_CRC.dr32().write_value(word);
```

Comment:
At least on STM32G4, any write to dr32 writes _four_ bytes into the CRC (hence the suggestive name). (Don't ask how long it took for me to figure this out... the reference manual hardly explains it at all :tired_face:)

If only a single byte write is intended, I suggest to write `PAC_CRC.dr8()` instead. If a four-byte write is intended, then the comment 1 line above should be updated.

Reply:
Good catch will rework - unfortunately there are other breaking changes (inc. tests) which need to be fixed all together to get a successful CI run - I'm not sure if I find the time to address them.

Reply:
To me it looks more like a question if that comment is correct - the function is called `feed_word()` why should it write a byte? It takes a `u32` - a word. Shouldn't we have another function to write a byte?

Reply:
Indeed, the comment is not consistent with the name of the function. Looking at crc/v2v3.rs, there are separate functions for `feed_word()` and `feed_byte()`.